### PR TITLE
Fixed #1654: Fix mknfold using new StratifiedKFold API

### LINKED ISSUE #1654: Look into error introduced by #1636
Issue tracker so I don't forget when I get back.

#1636  

cc: @tqchen

### PR DESCRIPTION
**Pull request by terrytangyuan**
Fixed #1654 @tqchen @CodingCat

### CODE REVIEW

**Comment by terrytangyuan** [comment]
Java flaky test: https://travis-ci.org/dmlc/xgboost/jobs/167141727 
@CodingCat Could you take a look?

**Comment by tqchen** [comment]
I am going to merge this in one. wait @CodingCat to fix the java testcase

**Comment by CodingCat** [comment]
Hmmm....this has appeared for multiple times, 

I will deprecate this test temporarily and figure out the reason later

**Comment by jokari69** [comment]
this commit makes it impossible to pass K Fold as an nfold argument (see comment in code). Please revert or add possibility to not shuffle (PR https://github.com/dmlc/xgboost/pull/1459)

**Comment by terrytangyuan** [comment]
Could you elaborate? I don't fully understand. There's no comment in the code either

**Comment by jokari69** [comment]
line 235 you added the check to see if the folds are an instance of list.
My end goal is that i don't want to shuffle my date so that is why i made the previous PR. My current work around was to pass in a KFold on which i set shuffle=False but this doesn't work anymore now.
My data is not iid and shuffling might create data leakage.

File: python-package/xgboost/training.py [hunk 1]
```diff
@@ -232,14 +232,12 @@ def mknfold(dall, nfold, param, seed, evals=(), fpreproc=None, stratified=False,
         randidx = np.random.permutation(dall.num_row())
         kstep = int(len(randidx) / nfold)
         idset = [randidx[(i * kstep): min(len(randidx), (i + 1) * kstep)] for i in range(nfold)]
-    elif folds is not None:
+    elif folds is not None and isinstance(folds, list):
```
Comment (terrytangyuan) [inline python-package/xgboost/training.py]: I commented on your PR. Could you address this in your PR too?

Comment (jokari69) [inline python-package/xgboost/training.py]: Sure, thanks for the quick reply

**Pull request merged**